cunumeric/eager.py

```diff
@@ -171,25 +191,31 @@ class EagerArray(NumPyThunk):
     :meta private:
     """
 
-    def __init__(self, runtime, array, parent=None, key=None) -> None:
+    def __init__(
+        self,
+        runtime: Runtime,
+        array: Any,
+        parent: Optional[Any] = None,
+        key: Optional[Any] = None,
```

Comment:
This could be further refined as "None or a tuple with a string at position 0 and anything else on positions 1+", but I don't think mypy supports such a type. We could change the code to take the same information in a nested tuple, i.e. something of type `tuple[str,tuple[Any, ...]]`, which is within the capabilities of mypy.

Reply:
Sure but I'd prefer to keep this first PR as documentation of what currently exists (as best as possible) and leave actual code/implementation changes to dedicated PRs

Reply:
Yes, I'm leaving this comment as documentation for future PRs